File: eureka/android/library/src/main/java/com/lifeplus/Pojo/Responses/ResponseBase.java
package com.lifeplus.Pojo.Responses;

import com.lifeplus.Pojo.Enum.ResultCodeEnum;

import java.util.HashMap;

public class ResponseBase {
    protected final ResultCodeEnum _resultCode;
    protected final String _message;
    protected final String _simuId;

    protected ResponseBase(ResultCodeEnum pErrorCode) {
        _resultCode = pErrorCode;
        _message = "";
        _simuId = "";
    }

    protected ResponseBase(ResultCodeEnum pErrorCode, String pMessage) {
        _resultCode = pErrorCode;
        _message = (pMessage != null) ? pMessage : "";
        _simuId = "";
    }

    protected ResponseBase(String pSimuId, String pSimuMessage) {
        _resultCode = null;
        _message = (pSimuMessage != null) ? pSimuMessage : "";
        _simuId = (pSimuId != null) ? pSimuId : "";
    }

    protected HashMap<String, Object> getResponseMap() {
        HashMap<String, Object> result = new HashMap<>();
        result.put( "status", (_resultCode != null) ? _resultCode.getType() : "failed" );
        result.put( "result", ((_simuId.isEmpty()) ? ((_resultCode != null) ? _resultCode.getCode() : "") : _simuId));
        result.put( "message", ((_simuId.isEmpty()) ? ((_resultCode != null) ? _resultCode.getCode() + ": " + _resultCode.getDesc() + ((_message.isEmpty()) ? "" : " (" + _message + ")") : _message ) : _message));

        return result;
    }

    public String getResponseStr() {
        return "";
    }

    public ResultCodeEnum getResultCode() {
        return _resultCode;
    }
}

File: eureka/ios/nativeLib/nativeLib/Models/Responses/ResponseBase.swift
//
//  ResponseBase.swift
//  BLEDemo
//

import Foundation

public class ResponseBase {
    
    internal final var _resultCode: ResultCodeEnum!
    internal final var _message: String
    internal final var _simuId: String
  
  internal init(pResponseBase:ResponseBase) {
    _resultCode = pResponseBase._resultCode
    _message = pResponseBase._message
    _simuId = pResponseBase._simuId
  }
  
  internal init(pErrorCode: ResultCodeEnum) {
    _resultCode = pErrorCode
    _message = ""
    _simuId = ""
  }
    
  internal init(pErrorCode: ResultCodeEnum, pMessage: String) {
    _resultCode = pErrorCode
    _message = pMessage
    _simuId = ""
  }

  internal init(pSimuId: String, pSimuMessage: String) {
    _resultCode = nil
    _message = pSimuMessage
    _simuId = pSimuId
  }
    
  internal func getResponseMap() -> [String: Any] {
    var result: [String : Any] = [:]
    result["status"] = (_resultCode != nil) ? _resultCode.type : "failed"
    result["result"] = (_simuId.isEmpty) ? ((_resultCode != nil) ? _resultCode.code : "") : _simuId
    result["message"] = (_simuId.isEmpty) ? ((_resultCode != nil) ? _resultCode.desc + ((_message.isEmpty) ? "" : " (" + _message + ")") : _message ) : _message
    return result
  }
    
  internal func getResponseStr() -> String {
    return ""
  }

  public func getResultCode() -> ResultCodeEnum {
    return _resultCode
  }
}
